### DISCUSSION TITLE
Github workflow_dispatch

### DISCUSSION THREAD

**Discussion by revoltig4sg** [community/community#63092] [Actions]
### Select Topic Area

Question

### Body

Can we have a feature in the GitHub actions to populate the workflow_dispatch input values dynamically, maybe using some API response or using other GitHub workflow outputs?

Ex: Current workflow
on:
workflow_dispatch:
inputs:
SOURCE_REGION:
type: choice
default:
required: true
options:
- us-west-1
- us-west-2
- us-east-1
- eu-west-1

Ex: Feature looking for-
on:
workflow_dispatch:
inputs:
SOURCE_REGION:
type: choice
default: [ SOME FALLBACK INPUT ]
required: true
options:
- {{ API_RESPONSE[0] }}
- {{ API_RESPONSE[1] }}
- {{ API_RESPONSE[2] }}
- {{ API_RESPONSE[3] }}

**Comment by Phoenix-Genius** [comment]
This (or something like it) is really important to support dynamic environments. It's getting pretty common to spin up environments for a single branch or testing of a release. Esp with microservices it can become a lot of work to update N number of workflow dispatch input choices.

While it could be environment variables that would be limiting and in a way bring one back to the same issue. Having to update data somewhere. One thing that might help a lot of users is to support a dynamic environment field in the workflow_dispatch that was dynamically generated from environments declared at the repo level.

**Comment by queenofcorgis** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
